src/client/updateClientMetaInfo.js

```diff
@@ -1,64 +1,75 @@
-import updateTitle from './updaters/updateTitle'
-import updateTagAttributes from './updaters/updateTagAttributes'
-import updateTags from './updaters/updateTags'
+import { updateAttribute, updateTag, updateTitle } from './updaters'
 
-export default function _updateClientMetaInfo (options = {}) {
+const getTag = (tags, tag) => {
+  if (!tags[tag]) {
+    tags[tag] = document.getElementsByTagName(tag)[0]
+  }
+
+  return tags[tag]
+}
+
+/**
+ * Performs client-side updates when new meta info is received
+ *
+ * @param  {Object} newInfo - the meta info to update to
+ */
+export default function updateClientMetaInfo(options = {}, newInfo) {
   const { ssrAttribute } = options
 
-  /**
-   * Performs client-side updates when new meta info is received
-   *
-   * @param  {Object} newInfo - the meta info to update to
-   */
-  return function updateClientMetaInfo (newInfo) {
-    const htmlTag = document.getElementsByTagName('html')[0]
-    // if this is not a server render, then update
-    if (htmlTag.getAttribute(ssrAttribute) === null) {
-      // initialize tracked changes
-      const addedTags = {}
-      const removedTags = {}
+  // only cache tags for current update
+  const tags = {}
+
+  const htmlTag = getTag(tags, 'html')
+
+  // if this is not a server render, then update
+  if (htmlTag.getAttribute(ssrAttribute) === null) {
+    // initialize tracked changes
+    const addedTags = {}
+    const removedTags = {}
+
+    Object.keys(newInfo).forEach((type) => {
```

Comment:
If babel is in, can we use `for in` here?

Reply:
@pi0 Whats the benefit of `for in` compared to `.forEach`? :thinking:

Reply:
I understood that `for in` is notorious for having a very big polyfill

Reply:
@manniL Code Readability + More ES6 + less mem usage when we use untranspiled version. `Object.keys` create a new Array from keys.

Reply:
PS: We use it in other areas like this:

https://github.com/nuxt/vue-meta/pull/319/files#diff-cba0880c28dd98835c4c104cb1467d57R13

Reply:
`for in` is in the standard for long. Do you mean `for of` (introduced with ES6 or so)? 🤔

Reply:
Ah yes, I might have. The object.keys have been factored out